src/driver.ts

```diff
@@ -339,10 +349,22 @@ export class AppiumFlutterDriver extends BaseDriver<FlutterDriverConstraints> {
          flutterServerLaunchTimeout:
             this.internalCaps?.flutterServerLaunchTimeout || 5000,
       } as DriverCaps<FlutterDriverConstraints>;
-      // @ts-ignore
-      const activateAppResponse = await this.proxydriver.activateApp(
-         appId || bundleId,
-      );
+      let activateAppResponse;
+      //run only for ios
+      if (
+         this.proxydriver instanceof XCUITestDriver &&
+         this.proxydriver.isSimulator() &&
+         this.internalCaps?.flutterSystemPort
+      ) {
+         activateAppResponse = await this.proxydriver.activateApp(
```

Comment:
I am not sure this is needed. Adjusting arguments in the xcuitest driver capabilities should be enough as it should also take care about starting the app. I would expect the app to be unexpectedly restarted after this call, which would break customer tests as its original env and args would be lost.

Reply:
@mykola-mokhnach The reason to do this is because once the activateApp from the customer tests is called we need to wait for the flutter server to start and create the session at flutter server. 
Is there any other way we could handle this?

`original env and args would be lost` Agree! We should handle this.

Reply:
#34 Issue created

Reply:
> The reason to do this is because once the activateApp from the customer tests is called we need to wait for the flutter server to start and create the session at flutter server.
Is there any other way we could handle this?

If we "instrument" xcuitest capabilities with proper args then the app should already start at the requested port. If the port is not provided explicitly then we just check the default one. The case If xcuitest has started with noReset option and the app was already running should be to document, so customers could explicitly provide the port number the build-in app server is running on if it is different from the default one.

Reply:
@mykola-mokhnach Agree, if user provides flutterSystemPort in caps the flutter driver will start the app and wait for the flutter server on that port to be reachable this is handled in the createSession of the flutter driver. 


User does
1. RemoveApp
2. InstallApp
3. activateApp 

Now when user calls activateApp with bundleId my understanding is that app should be launched with flutterSystemPort args and needs to be waited?? 

User does 
1. Just activateApp - in this case as the app is already running in the background on specific port xcuitest driver should have all the required information and we handle the way you have mentioned.

Reply:
If the user restarts the app then it is their responsibility to adjust the port number. 
In 95% of cases the port will be the default one. In the other 5%, where tests are executed in parallel or some other custom scenarios are applied there should be a document that describes what to do. 
We may also think about flutter specific API extensions for launching the app, which should be called instead of xcuitest/uia2 ones (e.f. `flutter: launchApp` instead of `mobile: launchApp`), which would respect the original port value and pass the corresponding arg to the app under test.

Reply:
I feel this idea make sense so we don't worry about xcui/uia2 implementation